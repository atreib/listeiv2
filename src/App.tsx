import React, { Suspense } from 'react';
import styled from 'styled-components';
import { StylesProvider } from '@material-ui/core/styles';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { NotFoundPage } from './components/pages';
import { Loading } from './components/utils';
import { AppRoutes, RoutesModel } from './routes';
import { ShoppingListProvider, ShoppingListHistoryProvider } from './contexts';
import { AppLayout } from './components/layout';

const ContainerDiv = styled.div`
  height: 100%;
`;

function App() {
  return (
    <StylesProvider injectFirst>
      <ContainerDiv className="App">
        <ShoppingListHistoryProvider>
          <ShoppingListProvider>
            <Router>
              <AppLayout>
                <Suspense fallback={<Loading>Carregando...</Loading>}>
                  <Switch>
                    {AppRoutes &&
                      AppRoutes.map((route: RoutesModel) => (
                        <Route key={route.path} path={route.path} exact component={route.component} />
                      ))}
                    <Route component={NotFoundPage} />
                  </Switch>
                </Suspense>
              </AppLayout>
            </Router>
          </ShoppingListProvider>
        </ShoppingListHistoryProvider>
      </ContainerDiv>
    </StylesProvider>
  );
}

export default App;
