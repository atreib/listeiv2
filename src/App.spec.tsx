import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('Loading component test suite', () => {
  it('Should render correctly', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
