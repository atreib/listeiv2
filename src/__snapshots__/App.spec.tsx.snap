// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component test suite Should render correctly 1`] = `
<styled.div
  className="App"
>
  <BrowserRouter>
    <Suspense
      fallback={
        <Loading>
          Carregando...
        </Loading>
      }
    >
      <Switch>
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_init": [Function],
              "_payload": Object {
                "_result": [Function],
                "_status": -1,
              },
            }
          }
          exact={true}
          key="/"
          path="/"
        />
        <Route
          component={[Function]}
        />
      </Switch>
    </Suspense>
  </BrowserRouter>
</styled.div>
`;
